src/framework/theme/components/stepper/stepper.component.ts

```diff
@@ -165,20 +186,23 @@ export class NbStepperComponent {
     this.steps.forEach(step => step.reset());
   }
 
-  isStepSelected(step: NbStepComponent) {
-    return this.index === this.steps.toArray().indexOf(step);
+  isStepSelected(step: NbStepComponent): boolean {
+    return this.selectedIndex === this.getStepIndex(step);
   }
 
-  private isStepValid(index: number): boolean {
-    const steps = this.steps.toArray();
+  isStepCompleted(step: NbStepComponent): boolean {
+    return !this.isStepSelected(step) && step.completed;
+  }
 
-    steps[this.index].interacted = true;
+  selectStep(step: NbStepComponent) {
+    step.select();
+  }
 
-    if (index >= this.index && index > 0) {
-      const currentStep = steps[this.index];
-      return currentStep.completed;
-    }
+  private getStepIndex(step: NbStepComponent): number {
+    return this.steps.toArray().indexOf(step);
+  }
 
-    return true;
+  private visitStep() {
```

Comment:
I'd personally prefer to name this function as `visitSelectedStep` because it just visits the selected step. What do you think?